src/effects/Equalization.cpp

```diff
@@ -430,24 +440,20 @@ bool EffectEqualization::Process(EffectInstance &, EffectSettings &)
             auto idealBlockLen = pChannel->GetMaxBlockSize() * 4;
             if (idealBlockLen % L != 0)
                idealBlockLen += (L - (idealBlockLen % L));
-            auto pNewChannel = pChannel->EmptyCopy();
-            pNewChannel->ConvertToSampleFormat(floatSample);
+            auto pNewChannel = *iter0++;
             Task task{ M, idealBlockLen, *pNewChannel };
-            if (ProcessOne(task, count, *pChannel, start, len)) {
-               temp->Add(pNewChannel);
-               assert(pNewChannel->IsLeader() == pChannel->IsLeader());
-            }
-            else
-               bGoodResult = false;
+
+            bGoodResult = ProcessOne(task, count, *pChannel, start, len);
+            if (!bGoodResult)
+               goto done;
```

Comment:
I don't have a strong opinion, but probably two breaks would improve this code a bit